webapp/src/dogma/features/project/settings/credentials/DeleteCredential.tsx

```diff
@@ -0,0 +1,67 @@
+import {
+  Button,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { newNotification } from 'dogma/features/notification/notificationSlice';
+import ErrorMessageParser from 'dogma/features/services/ErrorMessageParser';
+import { useAppDispatch } from 'dogma/hooks';
+import { MdDelete } from 'react-icons/md';
+
+export const DeleteCredential = ({
+  projectName,
+  id,
+  deleteCredential,
+  isLoading,
+}: {
+  projectName: string;
+  id: string;
+  deleteCredential: (projectName: string, id: string) => Promise<void>;
+  isLoading: boolean;
+}): JSX.Element => {
+  const { isOpen, onToggle, onClose } = useDisclosure();
+  const dispatch = useAppDispatch();
+  const handleDelete = async () => {
+    try {
+      await deleteCredential(projectName, id);
+      dispatch(newNotification('Credential deleted.', `Successfully deleted ${id}`, 'success'));
+      onClose();
+    } catch (error) {
+      dispatch(newNotification(`Failed to delete ${id}`, ErrorMessageParser.parse(error), 'error'));
+    }
+  };
+  return (
+    <>
+      <Button leftIcon={<MdDelete />} colorScheme="red" size="sm" variant="ghost" onClick={onToggle}>
+        Delete
+      </Button>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Are you sure?</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            Delete credential &apos{id}&apos from {projectName}?
```

Comment:
`&apos` is exposed without decoding.
```suggestion
            Delete credential '{id}'from {projectName}?
```
<img width="406" alt="image" src="https://github.com/user-attachments/assets/c03e2815-dd86-438c-a987-7dde480bdbf5">

Reply:
The IDE tricked me. 😓